components/eam/src/physics/cam/shoc_intr.F90

```diff
@@ -81,11 +82,18 @@ module shoc_intr
       shoc_ice_sh = 50.e-6  
          
   logical      :: lq(pcnst)
-  logical      :: lq2(pcnst)
+
+  !lq_dry_wet_cnvr is true for all the water constituents used  by SHOC.
+  !These water constituents will participate in dry/wet mmr conversion
+  logical      :: lq_dry_wet_cnvr(pcnst)
  
   logical            :: history_budget
   integer            :: history_budget_histfile_num  
   logical            :: micro_do_icesupersat
+
+  !Store names of the water constituents used by SHOC
```

Comment:
We probably need a logical here to switch between P3 and MG2.  I know that in SCREAM we only are committed to use P3, but in case in the future someone wants to experiment with MG2+SHOC this will throw an error or cause other issues since some of these constituents are not in MG2.